pkg/gatewayserver/scheduling/scheduler.go

```diff
@@ -102,21 +102,21 @@ func NewScheduler(
 		scheduleAnytimeDelay = &ScheduleTimeShort
 	}
 
-	var toa *frequencyplans.TimeOffAir
+	var timeOffAir *frequencyplans.TimeOffAir
 	for _, fp := range fps {
-		if toa != nil && fp.TimeOffAir != *toa {
+		if timeOffAir != nil && fp.TimeOffAir != *timeOffAir {
 			return nil, errFrequencyPlansTimeOffAir.New()
 		}
-		toa = &fp.TimeOffAir
+		timeOffAir = fp.TimeOffAir.Clone()
```

Comment:
Since `toa` was a pointer, clamping it with `QueueDelay` below actually modified the frequency plans.